### DISCUSSION TITLE
I'd like to ask you about the error.

### DISCUSSION THREAD

**Discussion by yukudaa** [langroodi/Adaptive-AUTOSAR#27] [Q&A]
![image](https://github.com/langroodi/Adaptive-AUTOSAR/assets/50691043/660525f9-faef-4aa5-a291-c108a6b113a7)

Hello. I'm doing it myself. 
But I'm asking because there's a difficult part. 
I get this error after writing api key and access token, can you tell me?

error message:
Setting the VIN failed due to unexpected RESTful response format.
TelematicControlModuleEvent is failed with DTC 49560
Alive supervision is expired on MachineFG

**Comment by langroodi** [comment]
Hi there! The only place you get `Setting the VIN failed due to unexpected RESTful response format.` is the following:

https://github.com/langroodi/Adaptive-AUTOSAR/blob/6d9b047be3b411168389894274fda4752c49d9e5/src/application/extended_vehicle.cpp#L210

It means your RESTful response from Volvo API does not contain `Vehicle` nor `Error` key. So, you need to put a breakpoint there while running at debug mode to check what JSON object you get there. At least what I can say is that it is not because of invalid authentication. Otherwise, you would get a response with `Error` key.

Accordingly, the VIN cannot be fetched that leads to the failure of the `ExtendedVehicle` adaptive application. That is why you get DTC _49560_. You can ignore `Alive supervision is expired on MachineFG`.

**Reply by yukudaa** [reply]
thank you!!

**Comment by yukudaa** [comment]
I'm so sorry... I tried putting breakpoints in debug mode but it didn't work out so can you tell me how to write them in ubuntu?

**Comment by langroodi** [comment]
No need to be sorry! Your response is vague! I did not get your point about _not working out_. It would be better to explain what you've done, what you've expected, and what actually happened. Also how do you mean by _writing them_? What are _them_?

Following is the image of VSCode IDE linked with C++ GDB Debugger running on Ubuntu (just as an example):
![image](https://github.com/langroodi/Adaptive-AUTOSAR/assets/19659758/f4c82ad5-d50b-4389-8e2d-ec40cf1282ad)
If you open the `extended_vehicle.cpp` on your repo via VSCode, and go to line 189 and press F9, you will see a red dot on the line that indicates that the breakpoint has been set. Then if you press the Debug icon on the status bar, you can run in debug mode (you don't need to pass the ARXML configuration in debug mode, because the application will fetch the default ones automatically).

Then you can go to the Terminal tab in VSCode to enter the RESTful authentication key. After a moment, the application should hit the breakpoint which happen by seeing a yellow arrow next to the red dot. Now, you can go to your Variable view and get the value of `_jsonResponse`.

If you do not have such a setup (i.e., IDE, Debugger, etc.) on your Ubuntu, you can find lots of tutorial on the Internet about get them ready and running. It is not something specific about Adaptive Autosar Simulator here.

**Reply by magic-tab-01** [reply]
if we got the value of JSON object, how to do next?

**Reply by langroodi** [reply]
Then I hope there will be something meaningful in HTTP response which you also can post here.

**Reply by test0-cmd** [reply]
Any updates about this issue, as i found it still exist
